src/permissionsAndroid.re

```diff
@@ -0,0 +1,217 @@
+
+  [@bs.module "react-native"]
+  [@bs.scope ("PermissionsAndroid", "PERMISSIONS")]
+  external readContacts : string = "READ_CONTACTS";
+
+  [@bs.module "react-native"]
+  [@bs.scope ("PermissionsAndroid", "PERMISSIONS")]
+  external writeContacts : string = "WRITE_CONTACTS";
+
+  [@bs.module "react-native"]
+  [@bs.scope ("PermissionsAndroid", "PERMISSIONS")]
+  external getAccounts : string = "GET_ACCOUNTS";
+
+  [@bs.module "react-native"]
+  [@bs.scope ("PermissionsAndroid", "PERMISSIONS")]
+  external accessFineLocation : string = "ACCESS_FINE_LOCATION";
+
+  [@bs.module "react-native"]
+  [@bs.scope ("PermissionsAndroid", "PERMISSIONS")]
+  external accessCoarseLocation : string = "ACCESS_COARSE_LOCATION";
+
+  [@bs.module "react-native"]
+  [@bs.scope ("PermissionsAndroid", "PERMISSIONS")]
+  external recordAudio : string = "RECORD_AUDIO";
+
+  [@bs.module "react-native"]
+  [@bs.scope ("PermissionsAndroid", "PERMISSIONS")]
+  external readPhoneState : string = "READ_PHONE_STATE";
+
+  [@bs.module "react-native"]
+  [@bs.scope ("PermissionsAndroid", "PERMISSIONS")]
+  external callPhone : string = "CALL_PHONE";
+
+  [@bs.module "react-native"]
+  [@bs.scope ("PermissionsAndroid", "PERMISSIONS")]
+  external readCallLog : string = "READ_CALL_LOG";
+
+  [@bs.module "react-native"]
+  [@bs.scope ("PermissionsAndroid", "PERMISSIONS")]
+  external writeCallLog : string = "WRITE_CALL_LOG";
+
+  [@bs.module "react-native"]
+  [@bs.scope ("PermissionsAndroid", "PERMISSIONS")]
+  external addVoicemail : string = "permission.ADD_VOICEMAIL";
+
+  [@bs.module "react-native"]
+  [@bs.scope ("PermissionsAndroid", "PERMISSIONS")]
+  external useSip : string = "USE_SIP";
+
+  [@bs.module "react-native"]
+  [@bs.scope ("PermissionsAndroid", "PERMISSIONS")]
+  external processOutgoingCalls : string = "PROCESS_OUTGOING_CALLS";
+
+  [@bs.module "react-native"]
+  [@bs.scope ("PermissionsAndroid", "PERMISSIONS")]
+  external bodySensors : string = "BODY_SENSORS";
+
+  [@bs.module "react-native"]
+  [@bs.scope ("PermissionsAndroid", "PERMISSIONS")]
+  external sendSms : string = "SEND_SMS";
+
+  [@bs.module "react-native"]
+  [@bs.scope ("PermissionsAndroid", "PERMISSIONS")]
+  external receiveSms : string = "RECEIVE_SMS";
+
+  [@bs.module "react-native"]
+  [@bs.scope ("PermissionsAndroid", "PERMISSIONS")]
+  external readSms : string = "READ_SMS";
+
+  [@bs.module "react-native"]
+  [@bs.scope ("PermissionsAndroid", "PERMISSIONS")]
+  external receiveWapPush : string = "RECEIVE_WAP_PUSH";
+
+  [@bs.module "react-native"]
+  [@bs.scope ("PermissionsAndroid", "PERMISSIONS")]
+  external receiveMms : string = "RECEIVE_MMS";
+
+  [@bs.module "react-native"]
+  [@bs.scope ("PermissionsAndroid", "PERMISSIONS")]
+  external readExternalStorage : string = "READ_EXTERNAL_STORAGE";
+
+  [@bs.module "react-native"]
+  [@bs.scope ("PermissionsAndroid", "PERMISSIONS")]
+  external writeExternalStorage : string = "WRITE_EXTERNAL_STORAGE";
```

Comment:
This is a bit verbose but I don't fear verbosity. This is a more succinct version:

```
   [@bs.module "react-native"]
   [@bs.scope ("PermissionsAndroid")]
   external permissions: Js.Dict.t(string) = "PERMISSIONS"
```

and then in `getPermission` you'd do `| READ_CALENDAR => Js.Dict.unsafeGet(permissions, "READ_CALENDAR")`